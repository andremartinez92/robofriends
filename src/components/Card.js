// @flow

import React from 'react';

type Props = {
  email: string,
  id: number,
  name: string,
};

function Card(props: Props) {
  const { email, id, name } = props;

  return (
    <div className="tc bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5">
      <img alt="robots" src={`https://robohash.org/${id}?200x200`} />
      <div>
        <h2>{name}</h2>
        <p>{email}</p>
      </div>
    </div>
  );
}

export default Card;
