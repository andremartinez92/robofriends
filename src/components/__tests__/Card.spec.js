import React from 'react';
import { shallow } from 'enzyme';

import Card from '../Card';

describe('<Card>', () => {
  const props = {
    email: 'test@test.com',
    id: 12,
    name: 'some name',
  };

  it('renders the expected component', () => {
    const subject = shallow(<Card {...props} />);
    expect(subject).toMatchSnapshot();
  });
});
