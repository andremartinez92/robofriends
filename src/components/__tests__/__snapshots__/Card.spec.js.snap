// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card> renders the expected component 1`] = `
<div
  className="tc bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5"
>
  <img
    alt="robots"
    src="https://robohash.org/12?size=200x200"
  />
  <div>
    <h2>
      some name
    </h2>
    <p>
      test@test.com
    </p>
  </div>
</div>
`;
